zkevm-circuits/src/evm_circuit/execution/extcodecopy.rs

```diff
@@ -53,17 +54,25 @@ impl<F: Field> ExecutionGadget<F> for ExtcodecopyGadget<F> {
 
     fn configure(cb: &mut EVMConstraintBuilder<F>) -> Self {
         let opcode = cb.query_cell();
+        let is_first_bytecode_table = cb.query_bool();
 
         let external_address_word = cb.query_word_rlc();
         let external_address =
             from_bytes::expr(&external_address_word.cells[..N_BYTES_ACCOUNT_ADDRESS]);
 
+        //let code_size = cb.query_cell();
         let memory_length = cb.query_word_rlc();
         let memory_offset = cb.query_cell_phase2();
+
         let code_hash = cb.query_cell_phase2();
         let not_exists = IsZeroGadget::construct(cb, code_hash.clone().expr());
         let exists = not::expr(not_exists.expr());
 
+        // the reason why not use `same_context.is_first_sub_bytecode` to construct `code_len_gadget` includes
+        // 1: `same_context` requires rw_counter_offset info, which depends on `code_offset` info, the latter relies on
+        // `code_len` of `code_len_gadget`
+        // 2: `same_context.is_first_sub_bytecode` returns current call bytecode info, in extcodecopy, code_hash is not
+        // necessary current call code hash.
```

Comment:
Are we missing the constraint that:
```
self::is_first_bytecode_table == code_len_gadget::is_first_bytecode_table
```

Reply:
found self::is_first_bytecode_table is useless and removed.